lib/bridge/MatrixHandler.js

```diff
@@ -369,7 +368,90 @@ MatrixHandler.prototype._onAdminMessage = Promise.coroutine(function*(req, event
         return;
     }
     else {
-        req.log.info("No valid admin command: %s", event.content.body);
+        let fmt = '\nirc.server\n'+
+                    '(COMMAND [arg0 [arg1 [...]]])\n'+
+                    '(COMMAND [arg0 [arg1 [...]]])\n'+
+                    'irc.server2';
+        req.log.info(`No valid (old form) admin command, will try new format`);
+
+        // Assumes commands have the form
+        // irc.server
+        // (COMMAND [arg0 [arg1 [...]]])
+        // (COMMAND [arg0 [arg1 [...]]])
+        // irc.server2
+        // ...
+        let commandBlocks = event.content.body;
+
+        let currentServer = ircServer;
+        let blacklist = ['PROTOCTL'];
+
+        try {
+            let lines = commandBlocks.replace("\r").split("\n");
+            for (var ix = 0; ix < lines.length; ix++) {
+                let line = lines[ix];
```

Comment:
I would almost certainly call `.trim()` on this so mobile clients don't curse your name as often software keyboards automatically insert spaces after words.